sources/web/datalab/polymer/editor.ts

```diff
@@ -38,21 +38,15 @@ if (editorElement) {
   // the url querystring
   editorElement.addEventListener('file-id-changed', (e: CustomEvent) => {
     const newId = e.detail.value as DatalabFileId;
-    if (newId) {
-      params.set('file', newId ? newId.toQueryString() : '');
-    } else {
-      params.delete('file');
-    }
-    const url = location.protocol + '//' + location.host + location.pathname +
-                '?' + decodeURIComponent(params.toString());
+    const url = location.protocol + '//' + location.host + '/editor/' + (newId || '');
     window.history.replaceState({}, '', url);
   });
 
   // Pass the file's path if it's specified in the location.
-  const params = new URLSearchParams(window.location.search);
-  if (params.has('file')) {
+  if (location.pathname.startsWith('/editor/')) {
```

Comment:
Might be worth adding something like "const prefix = '/editor/';" to avoid having the same string twice. Likewise for /notebook/ in the other file.

Reply:
Can we have resource-based URIs -- so `/notebooks/` and `/files/`?

Reply:
How does the user specify how to handle the resource though? Whether it should be downloaded, opened in the text editor, or opened in the notebook editor?